src/OpenLoco/Drawing/DrawSpriteHelper.hpp

```diff
@@ -5,9 +5,17 @@
 
 namespace OpenLoco::Drawing
 {
+    static inline uint8_t blend(const Gfx::PaletteMapView paletteMap, uint8_t src, uint8_t dst)
+    {
+        // src = 0 would be transparent so there is no blend palette for that, hence (src - 1)
+        assert(src != 0 && (src - 1u) < paletteMap.size());
```

Comment:
These asserts aren't correct anymore. Src-1 should be less than size/256.

Reply:
I cleaned up the entire function, now it should be clear.